cpp/src/binaryop/compiled/binary_ops.cuh

```diff
@@ -253,22 +258,52 @@ void for_each(rmm::cuda_stream_view stream, cudf::size_type size, Functor f)
 }
 
 template <class BinaryOperator>
-void apply_binary_op(mutable_column_device_view& outd,
-                     column_device_view const& lhsd,
-                     column_device_view const& rhsd,
+void apply_binary_op(mutable_column_view& out,
+                     column_view const& lhs,
+                     column_view const& rhs,
                      bool is_lhs_scalar,
                      bool is_rhs_scalar,
+                     binary_operator op,
                      rmm::cuda_stream_view stream)
 {
-  auto common_dtype = get_common_type(outd.type(), lhsd.type(), rhsd.type());
+  if (is_struct(lhs.type()) && is_struct(rhs.type())) {
+    auto op_order    = (binary_operator::GREATER == op || binary_operator::LESS_EQUAL == op)
+                         ? order::DESCENDING
+                         : order::ASCENDING;
+    auto flip_output = (binary_operator::GREATER_EQUAL == op || binary_operator::LESS_EQUAL == op);
```

Comment:
`binary_operator op` is not really required here.
These lines could use template argument `class BinaryOperator` to make these decisions.
With
```
template <class T, class... Ts>
inline constexpr bool is_any_v = std::disjunction<std::is_same<T, Ts>...>::value;
```
```suggestion
    auto op_order    = is_any_v<BinaryOperator, ops::Greater, ops::LessEqual>
                         ? order::DESCENDING
                         : order::ASCENDING;
    auto flip_output =  is_any_v<BinaryOperator, ops::GreaterEqual, ops::LessEqual>;
```